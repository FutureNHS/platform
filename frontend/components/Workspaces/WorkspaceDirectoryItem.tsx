import React from "react";

import Link from "next/link";
import styled from "styled-components";

const StyledImg = styled.img`
  display: block;
  height: 44px;
  width: 44px;
  border-radius: 4px;

  &:hover {
    & + div > a > h3 {
      ${({ theme }) => `
        color: ${theme.colorNhsukBlue};`}
      text-decoration: underline;
    }
  }
`;

const StyledContainer = styled.div`
  display: flex;
  margin-bottom: 40px;
`;

const StyledTextContainer = styled.div`
  margin: 0 25px;
`;

const StyledLink = styled.a`
  ${({ theme }) => `
    h3 {
      color: ${theme.colorNhsukBlack};
      &:hover {
        color: ${theme.colorNhsukBlue};
        text-decoration: underline;
        cursor: pointer;
      }

      &:active {
        color: ${theme.colorNhsukBlack};
        text-decoration: none;
        background-color: ${theme.colorNhsukYellow};
        border-bottom: 2px solid ${theme.colorNhsukBlack};
      }
 `}
`;

interface Props {
  title: string;
  id: string;
}

const WorkspaceDirectoryItem = ({ title, id }: Props) => {
  return (
    <StyledContainer>
      <StyledImg
        src={require("../../public/Placeholder_Workspace_Image.svg")}
        alt="https://www.nhs.co.uk"
      />
      <StyledTextContainer>
        <Link href="/workspaces/[id]" as={`/workspaces/${id}`}>
          <StyledLink>
            <h3>{title}</h3>
          </StyledLink>
        </Link>
      </StyledTextContainer>
    </StyledContainer>
  );
};

export default WorkspaceDirectoryItem;
