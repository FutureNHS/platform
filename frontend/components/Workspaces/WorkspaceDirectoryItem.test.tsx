import React from "react";

import { render } from "@testing-library/react";

import WorkspaceDirectoryItem from "./WorkspaceDirectoryItem";

test("snapshot of component", () => {
  const title = "workspace name";
  const id = "test123";

  const { asFragment } = render(
    <WorkspaceDirectoryItem title={title} id={id} />
  );

  expect(asFragment()).toMatchSnapshot();
});
