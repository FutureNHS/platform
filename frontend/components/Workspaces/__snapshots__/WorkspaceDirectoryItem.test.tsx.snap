// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of component 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 40px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 img {
  display: block;
  height: 44px;
  width: 44px;
  border-radius: 4px;
  cursor: pointer;
}

.c1 img:hover + h3 {
  color: undefined;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 h3 {
  margin-left: 25px;
  color: undefined;
}

.c1 h3:hover {
  color: undefined;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

.c1 h3:active {
  color: undefined;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: undefined;
  border-bottom: 2px solid undefined;
}

<div
    class="c0"
  >
    <a
      class="c1"
    >
      <img
        alt="https://www.nhs.co.uk"
        src="[object Object]"
      />
      <h3>
        workspace name
      </h3>
    </a>
  </div>
</DocumentFragment>
`;
